### PR TITLE
Add carousel to the main page [carousel-dev]

### PR DESCRIPTION
**Pull request by jakubcerveny**
Issue #106 - consider adding carousel to the main page.

I tried to select a couple of nice images with white background. We may not need all of them (and the captions may need tweaking).

The sentence "MFEM is a free, lightweight, ..." became one of the carousel captions, which is a little unorthodox...

**FIXME** Problems on smaller screen sizes.

Screenshot from 2021-05-21 13-07-19

### CODE REVIEW

**Comment by tzanio** [comment]
@tzanio will take a look

**Comment by tzanio** [comment]
Hi @jakubcerveny, 

I made some changes and fixes and I'm happy with the current state. Can you take another look and let me know if it works for you?

Do you want to make any further changes?

Tzanio

File: README.md [hunk 1]
```diff
@@ -4,17 +4,16 @@ This repo contains the MFEM website [MkDocs](http://www.mkdocs.org/) sources.
 
 To make changes to the website you will need an install of Python 3 with the following libraries:
 
-- use the latest versions of MkDocs, Markdown, PyYAML and mkdocs-exclude-search:
-  * `pip install --upgrade --user mkdocs`
-  * `pip install --upgrade --user Markdown`
+- use MkDocs v1.0.4 with Markdown v3.0 and the latest PyYAML and mkdocs-exclude-search:
+  * `pip install --upgrade --user mkdocs==1.0.4`
```
Comment (acfisher) [inline README.md:8]: Unfortunately this version breaks the search functionality.  I'll see if I can help get this working int he newer version.

Comment (tzanio) [reply README.md:8]: Hmm.. I specifically tried it and it worked for me. Can you do `pip show` on your versions?

Comment (acfisher) [reply README.md:8]: mkdocs = 1.0.4
markdown = 3.1.1
pyyaml = 5.4.1
mkdocs-exclude-search = 0.5.0

python = 2.7.15rc1

Comment (tzanio) [reply README.md:8]: Here are mine:

```
> pip show mkdocs Markdown PyYAML mkdocs-exclude-search | grep Version

Version: 1.0.4
Version: 3.0
Version: 5.4.1
Version: 0.5.2
```

Can you maybe try downgrading Markdown?

Comment (acfisher) [reply README.md:8]: That helped, but it is crashing on something in mkdocs-exclude-search.  For some reason I can't get 0.5.2 of that package.  Maybe I have an older version of python installed here?  What is your python version?

Comment (tzanio) [reply README.md:8]: I have multiple 😞 , but I think here I'm using 3.9

Comment (acfisher) [reply README.md:8]: I belive if you do a 'mkdocs --version' it will tell you which python you are running.

Comment (tzanio) [reply README.md:8]:
```
> mkdocs --version
mkdocs, version 1.0.4 from /Library/Python/3.9/lib/python/site-packages/mkdocs (Python 3.9)
```

Comment (acfisher) [reply README.md:8]: Interesting, when I install with mkdocs==1.0.4 using my python 3 it gives me mkdocs version 1.2

Comment (acfisher) [reply README.md:8]: I'm going to try upgrading my python3.

Comment (acfisher) [reply README.md:8]: Ok that fixed it.  I think we should recommend at least python 3.6.3

Comment (tzanio) [reply README.md:8]: Sorry for all the trouble @acfisher, hopefully mkdocs will be improved and we will be able to use the latest release in the future.

**Comment by jakubcerveny** [comment]
@tzanio, I like the changes, the carousel looks nice and it even seems to work on smaller screens, cool!

I tried building with mkdocs 1.2 and don't see any problems on the front page. Search seems to work.

**Comment by tzanio** [comment]
Cool, thanks @jakubcerveny!

@acfisher, do you think we should merge and deploy it?

**Review by acfisher — APPROVED** [review]
It looks like we got this working with the proper versions of everything so lets ship it.

**Pull request merged**